web/libs/datamanager/src/components/MainView/GridView/GridView.jsx

```diff
@@ -86,9 +90,8 @@ export const GridView = observer(({ data, view, loadMore, fields, onChange, hidd
   const renderItem = React.useCallback(
     ({ style, rowIndex, columnIndex }) => {
       const index = getCellIndex(rowIndex, columnIndex);
-      const row = data[index];
-
-      if (!row) return null;
+      if (index >= data.length || !data?.[index]) return null;
+      const row = data?.[index];
```

Comment:
are there any real worries that `data` will be missing? because if yes this code will still crash because of `data.length`. I see the need in these checks, but I'd suggest just to remove null checks.
```suggestion
      if (index >= data.length || !data[index]) return null;
      const row = data[index];
```

Reply:
```suggestion
      const row = data?.[index];
      if(!row) return null;
```